### DISCUSSION TITLE
Serial Number and row count in SQL Lab query results

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#18436] [Ideas]
**Is your feature request related to a problem? Please describe.**
Almost all popular SQL editors provide serial number(row number) and the total row count of results in the query results. 

**Describe the solution you'd like**
For row number, add a column to left-side with row number of query results.
- These numbers will not be used when user creates a table from query results clicking 'Explore', or downloads csv or copies results to clipboard.
- User cannot sort the results by the serial numbers.

For row count, the explore view already shows result row count in UI. We can show in similar way in SQL Lab as well.

some samples:
![image](https://user-images.githubusercontent.com/74634977/106974080-6eb4d700-6797-11eb-918c-74c7dde46de9.png)

![image](https://user-images.githubusercontent.com/74634977/106974116-7d9b8980-6797-11eb-969c-d87bfcc17976.png)

**Additional context**
This feature will bring Superset SQL editor a bit more closer to other popular editors, and make the users more comfortable using SQL lab more frequently.

**Comment by kamalkeshavani-aiinside** [comment]
Updated the feature request to show row count of query results as well.

**Comment by junlincc** [comment]
+1 on having total row count of results in SQL lab, should be a simple enhancement. I believe is in design phase 
https://github.com/apache/superset/issues/12436#issuecomment-771073296

Having serial number in the csv download file from Explore view is nice to have as well.

**Comment by Steejay** [comment]
Hey @kamalkeshavani-aiinside thanks for the feedback and feature requests! As @junlincc mentioned we're adding a row count indicator in SQL Lab for the results. Feedback is welcome on the designs posted in #12436. 

Re serial number. this is a great suggestion and would be a great result table enhancement. Im interested to learn more about this. Can you tell me about the last time you wanted to see serial numbers and why? What task were you trying to accomplish and how would serial numbers help you?

**Comment by kamalkeshavani-aiinside** [comment]
Hey @Steejay, here is a use case where I felt that serial numbers could have helped me recently.

- I was looking at a query result grouped by 2 columns, date and name(also ordered by date and name).
- You could say it is more of a habit from other SQL editors, where I can simply use serial number to find out how many rows with the first name are present in the results.
- In SQL lab, I just had to add row_number() in SELECT clause to get the same data.
 
I can't really say if a lot of people use the serial numbers the same way, but I am guessing that many of them have some usecase where they find it useful. And if I my guess is correct then I think this feature can help them feel more comfortable doing data exploration in SQL lab when starting to use Superset.
